hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -466,12 +627,90 @@ private Map<String, List<HoodieRecord<HoodieMetadataPayload>>> readAllLogRecords
     }
 
     try {
-      return logRecordReader.getAllRecordsByKeys(sortedKeys);
+      Map<String, List<HoodieRecord<HoodieMetadataPayload>>> records = logRecordReader.getAllRecordsByKeys(sortedKeys);
+      return records;
     } finally {
       timings.add(timer.endTimer());
     }
   }
 
+  private Map<String, List<HoodieRecord<HoodieMetadataPayload>>> readNonUniqueRecordsAndMergeWithLogRecords(HoodieSeekingFileReader<?> reader,
```

Comment:
This method is to merge the buffer obtained by reading all log-records with the buffer obtained by reading base-file.